### PR TITLE
TOOLS/CDVD: Replaced push_back, insert to emplace_*

### PR DESCRIPTION
**Pull request by GermanAizek**
### Description of Changes
@refractionpcsx2, Replaced insertion into a container with minor optimization and simpler code

### CODE REVIEW

File: pcsx2/DebugTools/ExpressionParser.cpp [hunk 1]
```diff
@@ -468,7 +468,7 @@ bool parsePostfixExpression(PostfixExpression& exp, IExpressionFunctions* funcs,
 					valueStack.push_back(val);
 				}
 				break;
-			case EXOP_SIGNPLUS:		// keine aktion n�tig
```
Comment (F0bes) [inline pcsx2/DebugTools/ExpressionParser.cpp:471]: Please discard this change

**Review by F0bes — CHANGES_REQUESTED** [review]

File: pcsx2/DebugTools/ExpressionParser.h [hunk 2]
```diff
@@ -18,8 +18,7 @@
 #include "common/Pcsx2Types.h"
 #include <vector>
 
-typedef std::pair<u64,u64> ExpressionPair;
-typedef std::vector<ExpressionPair> PostfixExpression;
+typedef std::vector<std::pair<u64,u64>> PostfixExpression;
```
Comment (stenzek) [inline pcsx2/DebugTools/ExpressionParser.h:21]: This is less clear to me than the original code.

Also, alias declarations (`using`) is preferable for new code.

**Comment by refractionpcsx2** [comment]
How is this optimising it? What exactly are you hoping to achieve by this? It seems like a pretty pointless change, like a "I just learned about this feature and think it's cool so I'm going to put it everywhere" kind of thing

**Comment by CookiePLMonster** [comment]
> How is this optimising it? What exactly are you hoping to achieve by this? It seems like a pretty pointless change, like a "I just learned about this feature and think it's cool so I'm going to put it everywhere" kind of thing

Admittedly in the case of pairs less type typing (pun unintended) is nice, but as for performance - it's a pair of two numbers, so the odds are it produces identical code anyways.

**Comment by stenzek** [comment]
> Could be beneficial for bigger types though because in-place construction of emplace entails less copying/moving.

Compiles to the same thing, at least for trivial types, which is what I expected. Complex types which are not default constructed might not.

https://godbolt.org/z/63ene63a9

**Comment by F0bes** [comment]
I'd think that any recruiter or company that looks at your profile will see that it is all artificially inflated. GitHub shouldn't be some weird place to min-max. There are multiple instances of your prs just being broken, this takes time away from maintainers. 
No thanks.

**Pull request closed without merging**